### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by AnnSerdechna**
[Demo]( https://annserdechna.github.io/react_redux-list-of-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,17 +1,94 @@
...
+  const dispatch = useDispatch();
+  const { loadTodos } = ACTIONS_CREATORS;
+
+  useEffect(() => {
+    dispatch(loadTodos());
+  }, []);
+
+  // useEffect(() => {
+  //   dispatch(loadUsers());
+  // }, []);
```
Comment (Ihor-Karpyn) [inline src/App.tsx:30]:
```suggestion
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,17 +1,94 @@
...
+  //     <div className="App__content-container">
+  //       {selectedUserId ? (
+  //         <UserDetails
+  //           userId={selectedUserId}
+  //         />
+  //       ) : 'No user selected'}
+  //     </div>
+  //   </div>
+  // </div>
+  );
```
Comment (Ihor-Karpyn) [inline src/App.tsx:76]: :harold

File: src/App.tsx [hunk 3]
```diff
@@ -1,17 +1,94 @@
...
+import { Finish } from './components/Finish';
+import { isLoading, getMessage, loadingTodos } from './store/selectors';
+import { getUsers } from './data/api';
+import { UserDetails } from './components/UserDetails';
+
+ *   const loading = useSelector(isLoading);
+  const message = useSelector(getMessage) || 'Ready!';
+
+ *     <div className="App">
       <h1>Redux list of todos</h1>
```
Comment (Ihor-Karpyn) [inline src/App.tsx:92]: remove comments

File: src/components/TodoItem/TodoItem.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+};
+
+export const TodoItem: FC<Props> = ({ todo }) => {
+  const selectedUserId = useSelector(getSelectedUserIdSelector);
+  const dispatch = useDispatch();
+
+  const { setSelectedUserId, removeTodoById } = ACTIONS_CREATORS;
+
+  const deleteTodo = async (todoId: number) => {
+    await removeTodo(todoId);
```
Comment (Ihor-Karpyn) [inline src/components/TodoItem/TodoItem.tsx:22]: check is comment deleted

File: src/components/TodoList/TodoList.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import { TodoItem } from '../TodoItem/TodoItem';
+import Form from '../TodoForm/TodoForm';
+import { Todo } from '../../types/todo.type';
+import './TodoList.scss';
+
+type Props = {
+  todos: Todo[];
+};
+
+export const TodoList: FC<Props> = memo(({ todos }) => {
```
Comment (Ihor-Karpyn) [inline src/components/TodoList/TodoList.tsx:15]: get todos using use selector hook, but not from props

File: src/components/UserDetails/UserDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import './UserDetails.scss';
+
+type Props = {
+  userId: number;
+};
+
+export const UserDetails: FC<Props> = memo(({
+  userId,
+}) => {
+  const [user, setUser] = useState<User | null>(Object);
```
Comment (Ihor-Karpyn) [inline src/components/UserDetails/UserDetails.tsx:18]: save user in redux

File: src/data/users.api.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import { User } from '../types/user.type';
+import { getData } from './api';
+import { ENDPOINTS } from './api.constants';
+
+export const getUsers = (): Promise<User[]> => getData(`${ENDPOINTS.users}`);
+
+export const getUser = (userId: number): Promise<User> => getData(`${ENDPOINTS.users}/${userId}`);
```
Comment (Ihor-Karpyn) [inline src/data/users.api.ts:7]:
```suggestion
export const getUserById = (userId: number): Promise<User> => getData(`${ENDPOINTS.users}/${userId}`);
```

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]

**Review by solaryasha — APPROVED** [review]

**Pull request closed without merging**